src/main/scala/beam/agentsim/events/PathTraversalEvent.scala

```diff
@@ -0,0 +1,141 @@
+package beam.agentsim.events
```

Comment:
I like this from a performance optimization perspective. I guess the only question is whether it's worth the gains (my guess is yes)? And more importantly, this is no longer flexible, where the old model I believe was more flexible to change of schema info and whatnot. Has that been discussed and agreed? If so, then maybe it's worth type-ifying all other areas that use getAttributes?

Reply:
We still need `getAttributes` to be able to write event to CSV/XML.